### ISSUE TITLE
Support "hoverable" for mobile targets as well

### ISSUE THREAD

**Issue by zoltish** [KoalaPlot/koalaplot-core#50]
I really dig the hoverable functionality on desktop. It would be sweet if something similar was available for Android and iOS. 

Im not sure which approach is most fitting for mobile, but I do love the tactile feeling of scrubbing through a graph and immediately seeing the closest value highlighted - either just above my finger, or at the TopCenter of the graph.

**Comment by gsteckman** [comment]
Yes, on Desktop the mouse cursor is always there so there is a hover concept. Any idea what would be the equivalent on Android or iOS - a press, a long press, or something else? Material3 has a TooltipBox, but last I checked it wasn't yet multiplatform. I expect it or a variant will eventually be made multiplatform at which time it might be possible to deprecate HoverableElementArea in favor of it. 

I think the feature you are describing is a little different than what currently exists in Koala Plot, where no matter where you click or hover on the y-axis a tooltip will display the y-axis values for the current x-axis position of the cursor. The current implementation requires a hover over a symbol to display the hover element.

**Comment by zoltish** [comment]
Sweet, I was not aware of TooltipBox! Looks like its CMP compatible as well 👍🏾 

Personal opinion obviously, but [heres a video](https://streamable.com/mcl9mm) from how the redesigned Fitbit app does it on Android. 

I think thats a great approach for a number of reasons, but perhaps most importantly - compared to clicking: I just need to tap the chart and I can kind of move through the points instead of having to specifically click a point (which Id say is problematic because theres no obvious surface that tells me that I can click it).

**Comment by zoltish** [comment]
> That video isn't available anymore. I think I know what you mean, but it would be great to see an example.

https://streamable.com/yrv51e

**Comment by pedrofsn** [comment]
Maybe we could use [this interface](https://developer.android.com/reference/android/view/View.OnHoverListener) for Android.

**Comment by michalgwo** [comment]
I managed to achieve part of what you need. A little modification is needed if you need to display value.

https://github.com/user-attachments/assets/463976db-61f5-4a40-abff-89d35a170c1b

``` kotlin
val touchX = remember { mutableFloatStateOf(0f) }
val isDragging = remember { mutableStateOf(false) }

AreaPlot(
    modifier = Modifier
        .drawWithContent {
            drawContent()
            if (isDragging.value) {
                drawCircle(
                    color = Color.Black,
                    radius = 15f,
                    center = findNearestPoint(
                        touchX = touchX.value,
                        values = data,
                        size = size
                    )
                )
            }
        }
        .pointerInput(Unit) {
            detectDragGestures(
                onDragStart = {
                    isDragging.value = true
                },
                onDrag = { change, _ ->
                    touchX.floatValue = change.position.x
                    change.consume()
                },
                onDragEnd = {
                    isDragging.value = false
                }
            )
    },
    data = data,
    lineStyle = LineStyle(brush = SolidColor(Color.Green), strokeWidth = 2.dp),
    areaStyle = AreaStyle(
        brush = SolidColor(Color.LightGray),
        alpha = 0.5f,
    ),
    areaBaseline = AreaBaseline.ConstantLine(0f)
)

private fun <X,Y>XYGraphScope<X,Y>.findNearestPoint(
    touchX: Float,
    values: List<DefaultPoint<X,Y>>,
    size: Size
): Offset {
    if (values.isEmpty())
        return Offset.Zero

    val lastIndex = values.lastIndex
    val maxX = scale(values.last(), size).x
    val minX = scale(values.first(), size).x
    val width = maxX - minX
    val constrainedTouchX = touchX.coerceIn(minX, maxX)

    val nearestPointIndex = (constrainedTouchX / width * lastIndex).toInt().coerceIn(0, lastIndex)
    val nearestPoint = scale(values[nearestPointIndex], size)

    val nextPointIndex = (nearestPointIndex + 1).coerceIn(0, lastIndex)
    val nextPoint = scale(values[nextPointIndex], size)

    val step = width / lastIndex
    val ratio = (constrainedTouchX % step) / step
    val interpolatedY = lerp(nearestPoint.y, nextPoint.y, ratio.coerceIn(0f, 1f))
    return Offset(constrainedTouchX, interpolatedY)
}
```

**Comment by gsteckman** [comment]
Nice. I think if the data points are not uniformly spaced the index calculation might return an incorrect result?

Since recently support was added for panning, which on Android requires a drag, I guess a plot can either support panning or support this effect, but not both. On desktop or other platforms with a mouse, both hover and drag interactions can be supported simultaneously.

**Comment by michalgwo** [comment]
> Nice. I think if the data points are not uniformly spaced the index calculation might return an incorrect result?

Yes, you are right, I couldn't find any way to properly convert the clicked x coordinate, to position on the chart, which would work in every case. Another way I found, but still not perfect, works only for numerical values, and judging by the source code, it can be wrong when panned or zoomed.
``` kotlin
val range = (xAxisModel as DoubleLinearAxisModel).range
val currentRange = range.start..(range.start + (range.endInclusive - range.start))
val x = (constrainedTouchX / size.width) * (currentRange.endInclusive - currentRange.start) + currentRange.start 
```

Is there any proper function for that purpose? What I'm looking for is the opposite of `scale()`.

**Comment by gsteckman** [comment]
Yes, that's a "missing" feature at this point, see also #55. It requires adding the inverse of ```computeOffset``` to the ```AxisModel``` interface, i.e., provide an offset between 0 and 1 and return the value in axis coordinate space.
